Removes Accidental IC in OOC

Incoming5643 opened the pull request:
OOC will catch messages that start with "." ";", ":", "#" (the say channel modifiers) or "say" itself.

When you type something that looks like IC in OOC you'll get a pop up that will let you actually say it if that's what you meant to do. It defaults to no so real accidental IC OOC can mash out of the command in case what they said was something like ";help I'm being killed!"

:cl: incoming5643
rscadd: OOC will now take steps to try and prevent accidental IC from people who mistakenly use OOC instead of say. This doesn't catch all mistakes, so please don't test it.
/:cl:

Incoming5643 committed:
OOC will now scrub out messages that start with . ; or : (the say channel modifiers and let the player know that they almost goofed up.

(if you want to make a fake IC in OOC joke you still can by adding a space before the message)

optimumtact commented:
GET THE FUCK OUT

Anonus commented:
# is also a say channel modifier, not that I've ever seen someone use it in ooc accidentally.

MrStonedOne commented:
Don't make it scrub it out, have it have a link the player can click to say it anyways or a pop up

having to add a space is bullshit

Incoming5643 committed:
Hashtag obscure functionality

PKPenguin321 commented:
add "say" to the list

Anonmare commented:
>No more tik-tok ick ock
Why do you hate fun?

optimumtact commented:
>(if you want to make a fake IC in OOC joke you still can by adding a space before the message)

Incoming5643 committed:
Preserves shitposting

when you type something that looks like IC in OOC you'll get a pop up that will let you actually say it if that's what you meant to do. It defaults to no so real accidental IC OOC can mash out of the command in case what they said was ";help!"

Incoming5643 commented:
Alright I've solved the problem in a way that lets you shitpost more effectively, happy?

Iamgoofball commented on code/game/verbs/ooc.dm:
```diff
@@ -14,6 +14,10 @@
 	if(!msg)	return
 
+	if((copytext(msg, 1, 2) in list(".",";",":","#")) || (lowertext(copytext(msg, 1, 4)) == "say"))
+		if(alert("Your message \"[msg]\" looks like it was meant for in game communication, say it in OOC?", "Meant for OOC?", "No", "Yes") != "Yes")
```
idea:
if you hit no, make it say() like you intended it to

Incoming5643 replied:
OOC is the same for everyone, but say is pretty dependent on what you are. I'm afraid.

MrStonedOne commented:
much better.

PKPenguin321 commented:
still needs "say" added to the list
please do this before i have to make another pr to do it once this is merged

Incoming5643 commented:
I put in say, scroll right

PKPenguin321 commented:
aha cool i'm blind
good job :+1:

MrStonedOne commented:
you could add asay and dsay if you really want to get jiggy with it.

Iamgoofball commented:
@MrStonedOne wouldn't the say one pick those up already

RemieRichards commented:
@Iamgoofball it's a string check for "say", not checking whether "say" occurs in the string, primarily because of stuff like using say as an actual word "What did he say?" I assume.

Dorsisdwarf commented:
![worthy](https://cloud.githubusercontent.com/assets/7305553/11398106/6d4a5ebc-9376-11e5-9a31-639401268beb.png)

WJohn commented:
:+1: x :100:

Robustin commented:
Yea I need this in my life.

Getting banned while im playing a head/antag for some mid-level IC shit by accident ain't cool.

Aranclanos commented:
>popup windows

jesus christ how horrifying

RemieRichards commented:
@Incoming5643 asay and dsay pls.

Incoming5643 committed:
adds functionality to catch dsay and asay mistakes, moves the check below things like OOC muting

Incoming5643 commented:
now checks for "say" within first four letters of message. This should catch "dsay" and "asay" as well as a few big mistypes like ";say". There aren't many words that <a href="http://www.wordfind.com/starts-with/say/">start with say</a> to false positive and there are NO words that have say as the second through forth letters.

Also moves the check below ooc mutes and such
